### PR TITLE
[core modules/doc] Update CONTRIBUTING.md for Move

### PR DESCRIPTION
**Pull request by tzakian**
Update the `CONTRIBUTING` doc for Move to include the new naming conventions, and also update comments and indentation within `script` and `address` blocks.

### CODE REVIEW

File: language/stdlib/CONTRIBUTING.md [hunk 1]
```diff
@@ -54,20 +54,66 @@ The functional testing framework is very convenient, but can't express all of th
...
-- If a module declares multiple resources/structs, the "main" type that clients of the module will use should be called `T` (this is obviously subjective).
-- Function names should be lowercase-underscore
-- Generic type parameter names should be long and descriptive (e.g., `deposit<Token>(t: Token)`). `T` should not be used as a generic type parameter name to avoid confusion with the resource/struct naming convention.
+- **Module names**: are camel case e.g., `LibraAccount`, `Libra`
+- **Type names**: are camel case e.g., `WithdrawalCapability`, `KeyRotationCapability`
+- **Function names**: are lower snake case e.g., `register_currency`
+- **Constant names**: are upper snake case e.g., `TREASURY_COMPLIANCE_ADDRESS`
+- Generic types should be descriptive, or anti-descriptive where appropriate (e.g. `T` for the Vector generic type parameter, `LibraAccount` for the core `LibraAccount` resource, `deposit<Token>(t: Token)` for depositing a token in the `Libra` module). Most of the time the "main" type in a module should be the same name as the module e.g., `Libra::Libra`, `LibraAccount::LibraAccount`.
```
Comment (sblackshear) [inline language/stdlib/CONTRIBUTING.md:61]: One question: do we want to standardize on `Token`, `CoinType`, or `Currency`? I think we use all three in the code right now...

File: language/stdlib/CONTRIBUTING.md [hunk 2]
```diff
@@ -54,20 +54,66 @@ The functional testing framework is very convenient, but can't express all of th
...
+- **Builtins** are unqualified and lower snake case e.g., `vector`, `borrow_global`
```
Comment (sblackshear) [inline language/stdlib/CONTRIBUTING.md:62]: Not a big deal, but not sure that this needs to be a part of the style guide given that contributors can't add new builtins...

File: language/stdlib/CONTRIBUTING.md [hunk 3]
```diff
@@ -54,20 +54,66 @@ The functional testing framework is very convenient, but can't express all of th
...
 - Each module, struct, resource, and public function declaration should be commented
-- Note: Move only has `//` comments (no block or doc comments like `///`, `//!` or `/* */`) at the moment
+- Move has both doc comments `///`, regular single-line comments `//`, and block comments `/* */`
+
 
 ## Formatting
 We plan to have an autoformatter to enforce these conventions at some point. In the meantime...
 
-- Four space identation
+- Four space identation except for `script` and `address` whose contents should not be indented
```
Comment (sblackshear) [inline language/stdlib/CONTRIBUTING.md:118]:
```suggestion
- Four space indentation except for `script` and `address` blocks whose contents should not be indented
```

File: language/stdlib/CONTRIBUTING.md (hunk 1 above)
Comment (tzakian) [reply language/stdlib/CONTRIBUTING.md:61]: ahhh yes. Good question, and I think we do want to standardize that. I am partial to `Currency`, but second preference is `CoinType`. Do you have a preference towards any of these three?

File: language/stdlib/CONTRIBUTING.md (hunk 2 above)
Comment (tzakian) [reply language/stdlib/CONTRIBUTING.md:62]: Yea..I wasn't sure whether to add it or not. I'll take it out in the name of being succinct.

**Pull request closed without merging**